### PR TITLE
feat(www): code hike

### PR DESCRIPTION
**Pull request by Isaiah-Hamilton**
## What kind of change does this PR introduce?

Moves code hike config and theme to shared config and also added to `www` just like this PR for the docs site: https://github.com/supabase/supabase/pull/14787

## What is the current behavior?

CleanShot 2023-06-06 at 00 33 10@2x

## What is the new behavior?

CleanShot 2023-06-06 at 00 33 00@2x

### CODE REVIEW

File: apps/www/code-hike.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { remarkCodeHike } from '@code-hike/mdx'
+
+declare module '@code-hike/mdx' {
+  export type CodeHikeRemarkPlugin = typeof remarkCodeHike
+  export type CodeHikeConfig = Parameters<CodeHikeRemarkPlugin>[0]
+}
```
Comment (Isaiah-Hamilton) [inline apps/www/code-hike.d.ts:6]: Could this also be moved to a shared config?

**Review by Isaiah-Hamilton — COMMENTED** [review]
Kind of throw the code hike configs into the `packages/config` folder. Should it have its own sub folder or stay how it is?

File: packages/config/code-hike.scss [hunk 2]
```diff
@@ -657,4 +657,4 @@
   -webkit-text-decoration-style: dotted;
   text-decoration-style: dotted;
   color: inherit;
-}
+}
```
Comment (gregnr) [inline packages/config/code-hike.scss:660]: Typically we keep the line ending on files
```suggestion
}

```

File: packages/config/code-hike.theme.json [hunk 3]
```diff
@@ -358,4 +358,4 @@
     "list.hoverBackground": "var(--ch-25)",
     "list.hoverForeground": "var(--ch-4)"
   }
-}
+}
```
Comment (gregnr) [inline packages/config/code-hike.theme.json:361]:
```suggestion
}

```

File: apps/docs/public/sitemap.xml [hunk 4]
```diff
@@ -4269,4 +4269,4 @@
     <changefreq>weekly</changefreq>
     <changefreq>0.5</changefreq>
   </url>
-</urlset>
+</urlset>
```
Comment (gregnr) [inline apps/docs/public/sitemap.xml:4272]:
```suggestion
</urlset>

```

**Review by gregnr — APPROVED** [review]
Nice 👍

File: apps/www/code-hike.d.ts (hunk 1 above)
Comment (gregnr) [reply apps/www/code-hike.d.ts]: Ideally yeah, but we can wait to do this with other type files too that should also be shared

**Comment by gregnr** [comment]
> Should it have its own sub folder or stay how it is?
This should be fine for now. There are lots of things that should be shared better, we can address these in a more dedicated task

File: apps/www/pages/_app.tsx [hunk 5]
```diff
@@ -14,6 +14,9 @@ import { post } from '~/lib/fetchWrapper'
 import { AuthProvider, ThemeProvider, useTelemetryProps } from 'common'
 import Head from 'next/head'
 
+import '@code-hike/mdx/dist/index.css'
```
Comment (gregnr) [inline apps/www/pages/_app.tsx:17]: QQ, was this line required by code hike?

Comment (Isaiah-Hamilton) [reply apps/www/pages/_app.tsx:17]: No must have left it there by mistake.

**Pull request merged**